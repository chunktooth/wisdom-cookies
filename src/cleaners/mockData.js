/* exported variableName */
export const mockQuotes = [
  {
    'message': 'Mankind is unkind, man',
    'id': '5280'
  },
  {
    'message': 'Godluck and Goodspeed!',
    'id': '1234'
  }
];