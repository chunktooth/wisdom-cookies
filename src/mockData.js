/* exported variableName */
export const mockWisdoms = [
  {
    'message': 'Mankind is unkind, man..',
    'id': '5280'
  },
  {
    'message': 'Godluck and Goodspeeeed!',
    'id': '1234'
  }
];

export const mockJar = [
  {
    'message': 'Mankind is unkind, man',
    'id': '5280'
  },
  {
    'message': 'Godluck and Goodspeed!',
    'id': '1234'
  }
];

export const mockMoonphases = { response: [
  { 
    timestamp: 1524433610, 
    dateTimeISO: "2018-04-22T21:46:50+00:00", 
    code: 1, 
    name: "first quarter"
  },
  {
    timestamp: 1525049954, 
    dateTimeISO: "2018-04-30T00:59:14+00:00", 
    code: 2, 
    name: "full moon"
  }]
};

export const mockMoonphasesArray = [{ response: [
  { 
    timestamp: 1524433610, 
    dateTimeISO: "2018-04-22T21:46:50+00:00", 
    code: 1, 
    name: "first quarter"
  },
  {
    timestamp: 1525049954, 
    dateTimeISO: "2018-04-30T00:59:14+00:00", 
    code: 2, 
    name: "full moon"
  }]
}]; 

export const mockMoonphase =  {
  date: "2018-04-22", 
  name: "first quarter"
};

export const mockWisdom = {
  'message': 'In Crust We Trust',
  'id': '1234'
};

export const mockCleanedMoons = [
  {
    date: "2018-04-22", 
    name: "first quarter"
  },
  { 
    date: "2018-04-30", 
    name: "full moon"
  }
];