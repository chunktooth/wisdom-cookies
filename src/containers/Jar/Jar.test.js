import React from 'react';
import { Jar,
  mapStateToProps } from './Jar';
import { shallow } from 'enzyme';
import { mockJar } from '../../mockData';

describe('Jar', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Jar jar={mockJar}/>);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should map jar state to props', () => {
    const mappedJar = mapStateToProps(mockJar);
    expect(mappedJar.jar).toEqual(mockJar.jar);
  });

});