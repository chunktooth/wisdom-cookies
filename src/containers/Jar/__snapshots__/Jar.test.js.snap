// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Jar should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    jar={
        Array [
            Object {
              "id": "5280",
              "message": "Mankind is unkind, man",
            },
            Object {
              "id": "1234",
              "message": "Godluck and Goodspeed!",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Connect(Component)
          id="5280"
          message="Mankind is unkind, man"
/>,
        <Connect(Component)
          id="1234"
          message="Godluck and Goodspeed!"
/>,
      ],
      "className": "wisdom-collection",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "5280",
        "nodeType": "class",
        "props": Object {
          "id": "5280",
          "message": "Mankind is unkind, man",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1234",
        "nodeType": "class",
        "props": Object {
          "id": "1234",
          "message": "Godluck and Goodspeed!",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Connect(Component)
            id="5280"
            message="Mankind is unkind, man"
/>,
          <Connect(Component)
            id="1234"
            message="Godluck and Goodspeed!"
/>,
        ],
        "className": "wisdom-collection",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "5280",
          "nodeType": "class",
          "props": Object {
            "id": "5280",
            "message": "Mankind is unkind, man",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1234",
          "nodeType": "class",
          "props": Object {
            "id": "1234",
            "message": "Godluck and Goodspeed!",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
